src/uid2DiiNormalization.ts

```diff
@@ -1,75 +1,65 @@
-enum EmailParsingState {
-  Starting,
-  SubDomain,
-}
-
 export function isNormalizedPhone(phone: string): boolean {
   return /^\+[0-9]{10,15}$/.test(phone);
 }
 
-export function normalizeEmail(email: string): string | undefined {
-  if (email == undefined || email.length <= 0) {
-    return undefined;
-  }
+const EMAIL_EXTENSION_SYMBOL = "+";
+const EMAIL_DOT = ".";
+const GMAIL_DOMAIN = "gmail.com";
 
-  let preSb = "";
-  let preSbSpecialized = "";
-  let sb = "";
-  let wsBuffer = "";
+type EmailParts = {
+  address: string;
+  domain: string;
+};
 
-  let parsingState = EmailParsingState.Starting;
-  let inExtension = false;
+function splitEmailIntoAddressAndDomain(email: string): EmailParts | undefined {
+  const parts = email.split("@");
+  if (!parts.length || parts.length !== 2 || parts.some((part) => part === ""))
+    return;
 
-  for (const cGiven of email) {
-    let c = cGiven;
-    if (cGiven >= "A" && cGiven < "Z") {
-      c = String.fromCharCode(c.charCodeAt(0) + 32);
-    }
+  return {
+    address: parts[0],
+    domain: parts[1],
+  };
+}
+
+function isGmail(domain: string): boolean {
+  return domain === GMAIL_DOMAIN;
+}
 
-    switch (parsingState) {
-      case EmailParsingState.Starting:
-        if (c == " ") {
-          break;
-        }
-        if (c == "@") {
-          parsingState = EmailParsingState.SubDomain;
-        } else if (c == ".") {
-          preSb += c;
-        } else if (c == "+") {
-          preSb += c;
-          inExtension = true;
-        } else {
-          preSb += c;
-          if (!inExtension) {
-            preSbSpecialized += c;
-          }
-        }
-        break;
-      case EmailParsingState.SubDomain:
-        if (c == "@") {
-          return undefined;
-        } else if (c == " ") {
-          wsBuffer += c;
-          break;
-        } else if (wsBuffer.length > 0) {
-          sb += wsBuffer;
-          wsBuffer = "";
-        }
-        sb += c;
-    }
-  }
+function dropExtension(
+  address: string,
+  extensionSymbol: string = EMAIL_EXTENSION_SYMBOL
+): string {
+  return address.split(extensionSymbol)[0];
+}
+
+function normalizeAddressPart(
+  address: string,
+  shouldRemoveDot: boolean,
+  shouldDropExtension: boolean
+): string {
+  let parsedAddress = address.replaceAll(" ", "");
```

Comment:
I don't think there can be any spaces by this point - do you need this check?

Reply:
Yes, there shouldn't be any spaces, removed the replaceAll, thanks